pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,192 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        scmd = 'merge'
+        ssp = sp.add_parser(scmd, help=f'{cmd} {scmd} --help')
+        ssp.set_defaults(process=TavgCLIPlugin.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+
+        # Open all the solition files
+        self._prepare_files(self.args.solns)
+
+        # Build up which files are needed and if std can be created
+        active = self._active_files()
+        merged_times = [(r.start, r.end) for r, f, s in active]
+        avg_time = sum([t[1] - t[0] for t in merged_times])
+        self.std_all = all([s == 'all' for r, f, s in active])
+
+        # Initialise things needed for the merge
+        r0, f0, s0 = active[0]
+        self._init_tavg_merge(*f0)
+
+        # Check all files are compatible for merge
+        self._compatibility_merge(active)
+
+        with h5py.File(self.args.output, 'w') as out:
+            # Merge the averages and stds
+            self._merge_data(out, active, avg_time)
+
+            # Merge runtime stats
+            self._merge_runtime_stats(active)
+
+            # Merge the metadata
+            self._merge_meta(out, merged_times)
+
+    def _active_files(self):
+        times = []
+        for file, stats, cfg in self.sfiles.values():
+            cfgsect = stats.get('tavg', 'cfg-section')
+            std = cfg.get(cfgsect, 'std-mode')
+            twindow = stats.getliteral('tavg', 'range')
+
+            for w in twindow:
+                times.append((Interval(*w), (file, stats, cfg), std))
+
+        active = times
+        for i, (r, f, s) in enumerate(times):
+            if any([(r in R and f != F) for R, F, S in active]):
+                active.pop(i)
+            elif any([r.o_intersection(R) for R, F, S in active]):
+                raise ValueError('Clash of average file time windows')
+
+        return active
+
+    def _compatibility_merge(self, active):
+        for t_rng, (file, stats, cfg), std_mode in active:
```

Comment:
`t_rng` may want to change the suffix as `rng` is often used for random number generators.  Maybe ival or seg or wind?